src/alr/alr-build_env.adb

```diff
@@ -0,0 +1,175 @@
+with Ada.Strings.Unbounded;
+with Ada.Text_IO;
+
+with GNAT.OS_Lib;
+
+with Alire.Utils;
+with Alire.Properties.Scenarios;
+with Alire.GPR;
+with Alire.Directories;
+
+with Alr.Platform;
+with Alr.Paths;
+with Alr.OS_Lib;
+with Alr.Query;
+with Alr.Commands;
+
+package body Alr.Build_Env is
+
+   type Env_Var_Action_Callback is access procedure (Key, Val : String);
+
+   Path_Separator : constant Character := GNAT.OS_Lib.Path_Separator;
+
+   ------------------
+   -- Project_Path --
+   ------------------
+
+   function Project_Path (Instance : Query.Instance;
+                          Root     : Alire.Roots.Root)
+                          return String
+   is
+      use Alr.OS_Lib;
+      use Ada.Strings.Unbounded;
+
+      Result       : Unbounded_String;
+      All_Paths    : Alire.Utils.String_Vector;
+      Sorted_Paths : Alire.Utils.String_Set;
+
+      Working_Folder : constant Alire.Absolute_Path :=
+        Alire.Directories.Current / Paths.Alr_Working_Folder;
+   begin
+      --  First obtain all paths and then output them, if any needed
+      for Rel of Instance.Including (Root.Release) loop
+         if Rel.Name = Root.Release.Name then
+            --  All_Paths.Append (".");
+            null; -- That's the first path in aggregate projects anyway
+         else
+            All_Paths.Append (Working_Folder /
+                                Paths.Alr_Working_Deps_Path /
+                                  Rel.Unique_Folder);
+         end if;
+
+         --  Add non-root extra project paths, always
+         for Path of Rel.Project_Paths (Platform.Properties) loop
+            All_Paths.Append
+              (Working_Folder /
+                 (if Rel.Name = Root.Release.Name
+                  then ".."
+                  else Paths.Alr_Working_Deps_Path / Rel.Unique_Folder) /
+                 Path);
+         end loop;
+      end loop;
+
+      --  Sort and remove duplicates in paths (may come from extension
+      --  projects).
+      for Path of All_Paths loop
+         Sorted_Paths.Include (Path);
+      end loop;
+
+      if not Sorted_Paths.Is_Empty then
+         for Path of Sorted_Paths loop
+            Result := Result & Path & Path_Separator;
```

Comment:
This leaves a trailing separator, although I guess it's harmless.

Reply:
Harmless indeed, I can still try to remove it.